app/javascript/components/advanced-settings-form/index.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import YAML from 'yaml';
+
+import MiqFormRenderer from '@@ddf';
+import createSchema from './advanced-settings-form.schema';
+
+const AdvancedSettingsForm = ({ url }) => {
+  const [{ isLoading, initialValues }, setState] = useState({ isLoading: true });
+
+  useEffect(() => {
+    API.get(`${url}/settings`).then((settings) => setState({
```

Comment:
Is generating api urls in JS absolutely necessary?

adding something to the output of `api_resource_path` makes the use of `api_resource_path` absolutely useless .. if we have a helper to return api urls, let's use **those** urls, without modifying them further

Reply:
Even better .. just use a switch with the *full* URLs. That way, no `api_resource_path` helper is needed, and we would actually know what kinds of entities this works on.

Reply:
Well, the helper can't deal with subcollection URLs...but feel free to extend it.

Reply:
I could possibly pass the `settings` in the `react` method in the view, but that's the best we can do so far :disappointed: